app/App.php

```diff
@@ -2,4 +2,52 @@
 
 declare(strict_types = 1);
 
-// Your Code
+function readCsv (): array{
+    $csvFile = fopen('../transaction_files/sample_1.csv', "r");
+    while ((($file = fgetcsv($csvFile, 100, ',')) !== false)) {
+        $array[] = $file;
+    }
+    fclose($csvFile);
+    unset($array[0]);
+    return $array;
+}
+
+function calculateTotalIncome (array $csvArray): array
+{
+    $totalIncome = 0.0;
+    $totalExpense = 0.0;
+    foreach ($csvArray as $line){
+        $value = (float) str_replace(array("$",","),"",$line[3]);
+        if ($value > 0){
+            $totalIncome += $value;
+        }else{
+            $totalExpense += $value;
+        }
+    }
+    $total[0] = $totalIncome;
+    $total[1] = $totalExpense * -1;
+    return $total;
+}
+
+function printDates (array $csvArray): void{
+
+    foreach ($csvArray as $line){
+//        $date = date_create ($line[0]);
+        echo "<tr>";
```

Comment:
I'd avoid echoing html from PHP this way. Better to pass data to view & have view be responsible for rendering the html